x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_enrollment_flyout/agent_policy_selection.tsx

```diff
@@ -33,131 +29,48 @@ type Props = {
     }
 );
 
+const resolveAgentId = (
+  agentPolicies?: AgentPolicy[],
+  selectedAgentId?: string
+): undefined | string => {
+  if (agentPolicies && agentPolicies.length && !selectedAgentId) {
+    if (agentPolicies.length === 1) {
+      return agentPolicies[0].id;
+    }
+
+    const defaultAgentPolicy = agentPolicies.find((agentPolicy) => agentPolicy.is_default);
+    if (defaultAgentPolicy) {
+      return defaultAgentPolicy.id;
+    }
+  }
+
+  return selectedAgentId;
+};
+
 export const EnrollmentStepAgentPolicy: React.FC<Props> = (props) => {
-  const { notifications } = useStartServices();
   const { withKeySelection, agentPolicies, onAgentPolicyChange, excludeFleetServer } = props;
   const onKeyChange = props.withKeySelection && props.onKeyChange;
-
-  const [isAuthenticationSettingsOpen, setIsAuthenticationSettingsOpen] = useState(false);
-  const [enrollmentAPIKeys, setEnrollmentAPIKeys] = useState<GetEnrollmentAPIKeysResponse['list']>(
-    []
+  const [selectedAgentId, setSelectedAgentId] = useState<undefined | string>(
+    () => resolveAgentId(agentPolicies, undefined) // no agent id selected yet
```

Comment:
Do we need this lazy initializer for this piece of state here? This _might_ be simpler if we removed the arrow function and passed in the result of `resolveAgentId`, but I want to make sure I'm not missing the intent here or discarding any concerns.

Reply:
Ah got it. I understand now. All good on my end 👍